src/solvers/refinement/string_refinement.cpp

```diff
@@ -1685,14 +1701,26 @@ exprt string_refinementt::get(const exprt &expr) const
     if(it!=found_length.end())
       return get_array(ecopy, it->second);
   }
+  else if(refined_string_typet::is_refined_string_type(ecopy.type()) &&
```

Comment:
It shouldn't be needed, because get should be able to handle `string_exprt` (it is a normal `struct_exprt`), arrays are treated differently because we want to set there length to the length of the string, it may not be totally necessary but it can make the result nicer.